src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -1007,7 +1017,20 @@ static inline bool notification_match(reservation_t& res,
         return ret;
       }
     }
-    res.topics.emplace_back(topic_filter.s3_id, topic_cfg, res_id);
+    // load the topic,if there is change in topic config while it's stored in
+    // notification.
+    rgw_pubsub_topic result;
+    const RGWPubSub ps(res.store, res.user_tenant);
+    auto ret = ps.get_topic(res.dpp, topic_cfg.name, result, res.yield);
```

Comment:
> we should check here that the user that does the operation is authorized to access the topic. similarly to how we do that in the notification creation command

This code is while delivering the nnotification, so by this code the user is already allowed, else it would not even allow to set notification !

Reply:
it is allowed to use notifications on the bucket, but not necessarily to access the topic

Reply:
> it is allowed to use notifications on the bucket, but not necessarily to access the topic

yeah you have different permission for accessing topic and allowing for publishing, but this code is about the loading the topic to deliver the notification and not about sending the topic to the user, so checking the permission here does not make sense